### PR TITLE
failsafe documentation update

### PR DESCRIPTION
**Pull request by frozenprocess**
* Corrects how to disable failsafe via Felix configurations
* Adds the current list of ports to the crds and files

## Related issues/PRs
Fixes https://github.com/projectcalico/calico/issues/8640

### CODE REVIEW

File: api/pkg/apis/projectcalico/v3/felixconfig.go [hunk 1]
```diff
@@ -344,20 +344,20 @@ type FelixConfigurationSpec struct {
...
-	// ports to ensure that Felix does not get cut off from etcd as well as allowing DHCP and DNS.
-	// [Default: tcp:179, tcp:2379, tcp:2380, tcp:6443, tcp:6666, tcp:6667, udp:53, udp:67]
+	// FailsafeOutboundHostPorts is a list of List of PortProto struct objects including UDP/TCP/SCTP ports and CIDRs that Felix
+	// will allow outgoing traffic from host endpoints to irrespective of the security policy. This is useful to avoid accidentally
+	// cutting off a host with incorrect configuration. For backwards compatibility, if the protocol is not specified, it defaults
+	// to "tcp". If a CIDR is not specified, it will allow traffic from all addresses. To disable all outbound host ports,
+	// use the value "[]". The default value opens etcd's standard ports to ensure that Felix does not get cut off from etcd
+	// as well as allowing DHCP, DNS, BGP and the Kubernetes API.
+	// [Default: {"port":53,"protocol":"udp"},{"port":67,"protocol":"udp"}, {"port":179,"protocol":"tcp"}, {"port":2379,"protocol":"tcp"}, {"port":2380,"protocol":"tcp"}, {"port":5473,"protocol":"tcp"}, {"port":6443,"protocol":"tcp"}, {"port": 6666,"protocol":"tcp"}, {"port":6667,"protocol":"tcp"}]
```
Comment (tomastigera) [inline api/pkg/apis/projectcalico/v3/felixconfig.go:360]: Is this really how it should be specified? Or should it be udp:53 and then itis parsed into the struct? HOw would you specify it in envvar?

Comment (frozenprocess) [reply api/pkg/apis/projectcalico/v3/felixconfig.go:360]: Judging by the test cases, I would say my approach is correct. 
https://github.com/projectcalico/calico/blob/fa5e216cd7af37c594a224d0afb070d3ccea0ff9/felix/config/config_params_test.go#L385C1-L389C6

Comment (caseydavenport) [reply api/pkg/apis/projectcalico/v3/felixconfig.go:360]: That's the Golang representation, but the yaml representation might be different. 

Not sure the right one to put here. Probably should be representation agnostic?

Comment (caseydavenport) [reply api/pkg/apis/projectcalico/v3/felixconfig.go:360]: Note this part of the test cases you linked:

> "tcp:0.0.0.0/0:1,udp:0.0.0.0/0:2"

^ This is the input, which is then parsed into the structs.

**Comment by caseydavenport** [comment]
@tomastigera @frozenprocess what's the status of this one?

**Comment by frozenprocess** [comment]
I've implemented your commnets and it was waiting for a merge.

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**